### PR TITLE
[SYCL] Add ESIMD test to check kernel arg size

### PR DESCRIPTION
**Pull request by srividya-sundaram**

### CODE REVIEW

File: clang/test/SemaSYCL/Inputs/sycl.hpp [hunk 1]
```diff
@@ -143,6 +143,7 @@ class __attribute__((sycl_special_class)) __SYCL_TYPE(accessor) accessor {
   using PtrType = typename DeviceValueType<dataT, accessTarget>::type *;
   void __init(PtrType Ptr, range<dimensions> AccessRange,
               range<dimensions> MemRange, id<dimensions> Offset) {}
+  void __init_esimd(PtrType Ptr) {}
```
[the remark this answers is not in the document]
Comment (elizabethandrews) [reply clang/test/SemaSYCL/Inputs/sycl.hpp:146]: Think it does.

File: clang/test/SemaSYCL/args-size-overflow.cpp [hunk 2]
```diff
@@ -11,9 +12,11 @@ queue q;
 using Accessor =
     accessor<int, 1, sycl::access::mode::read_write, sycl::access::target::global_buffer>;
 #ifdef SPIR64
-// expected-warning@27 {{size of kernel arguments (7994 bytes) may exceed the supported maximum of 2048 bytes on some devices}}
+// expected-warning@30 {{size of kernel arguments (7994 bytes) may exceed the supported maximum of 2048 bytes on some devices}}
+// expected-warning@31 {{size of kernel arguments (7988 bytes) may exceed the supported maximum of 2048 bytes on some devices}}
```
Comment (elizabethandrews) [inline clang/test/SemaSYCL/args-size-overflow.cpp:16]: I assume the difference in size is due to difference in __init and __init_esimd function arguments?

**Review by elizabethandrews — APPROVED** [review]

**Review by premanandrao — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

File: clang/test/SemaSYCL/args-size-overflow.cpp (hunk 2 above)
Comment (srividya-sundaram) [reply clang/test/SemaSYCL/args-size-overflow.cpp:16]: Yes, that is my understanding.

**Comment by dm-vodopyanov** [comment]
All failures are in SYCL E2E tests, unrelated to this patch.

**Pull request merged**